objects/src/accounts/builder/mod.rs

```diff
@@ -0,0 +1,436 @@
+    /// returns `None` for the seed, skips the grinding of an account id and constructs one
+    /// instead. Hence it is always preferable to use this method in testing code.
+    ///
+    /// For possible errors, see the documentation of [`Self::build`].
+    #[cfg(feature = "testing")]
+    pub fn build_testing(self) -> Result<(Account, Option<Word>), AccountError> {
+        let (init_seed, vault, code, storage) = self.build_inner()?;
+
+        let (account_id, seed) = if self.nonce == ZERO {
+            let (account_id, seed) =
+                self.grind_account_id(init_seed, code.commitment(), storage.commitment())?;
+
+            (account_id, Some(seed))
+        } else {
+            let account_id =
+                Self::construct_account_id(self.account_type, self.storage_mode, init_seed);
+
+            (account_id, None)
+        };
+
+        let account = Account::from_parts(account_id, vault, storage, code, self.nonce);
+
+        Ok((account, seed))
+    }
+}
+
+#[cfg(feature = "testing")]
+impl AccountBuilder {
+    /// Sets the nonce of the account. This method is optional.
+    ///
+    /// If unset, the nonce will default to [`ZERO`].
+    pub fn nonce(mut self, nonce: Felt) -> Self {
+        self.nonce = nonce;
+        self
+    }
+
+    /// Adds all the assets to the account's [`AssetVault`]. This method is optional.
+    ///
+    /// Must only be called when nonce is non-[`ZERO`] since new accounts must have an empty vault.
+    pub fn with_assets<I: IntoIterator<Item = Asset>>(mut self, assets: I) -> Self {
+        self.assets.extend(assets);
+        self
+    }
+
+    /// Constructs an [`AccountId`] for testing purposes with the given account type and storage
+    /// mode and using the first 8 bytes of the `init_seed` as part of the account id.
+    fn construct_account_id(
+        account_type: AccountType,
+        storage_mode: AccountStorageMode,
+        init_seed: [u8; 32],
+    ) -> AccountId {
+        let id_high_nibble = (storage_mode as u8) << 6 | (account_type as u8) << 4;
+
+        let mut bytes =
+            <[u8; 8]>::try_from(&init_seed[0..8]).expect("we have sliced exactly 8 bytes off");
+
+        // Clear the highest five bits of the most significant byte.
+        // The high nibble must be cleared so we can set it to the storage mode and account type
+        // we've constructed.
+        // The 5th most significant bit is cleared to ensure the resulting id is a valid Felt even
+        // when all other bits are set.
+        bytes[0] &= 0x07;
+        // Set high nibble of the most significant byte.
+        bytes[0] |= id_high_nibble;
+
+        let account_id = Felt::try_from(u64::from_be_bytes(bytes))
+            .expect("must be a valid felt after clearing the 5th highest bit");
+        let account_id = AccountId::new_unchecked(account_id);
+
+        debug_assert_eq!(account_id.account_type(), account_type);
+        debug_assert_eq!(account_id.storage_mode(), storage_mode);
```

Comment:
Should we move this logic into `AccountId` struct (i.e., create a new constructor there under `testing` feature).

Reply:
Sounds good! Moved it.